src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || size < index) {
+            throw new IndexOutOfBoundsException("The index " + index + " is invalid "
+                    + "the actual size of array is " + size);
+        }
+        Node<T> newNode = new Node<>(value);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. In this case, you should pass the references to the previous and next nodes as well.